modular_zubbers/code/modules/storyteller/scheduled_event.dm

```diff
@@ -0,0 +1,94 @@
+///Scheduled event datum for SSgamemode to put events into.
+/datum/scheduled_event
+	/// What event are scheduling.
+	var/datum/round_event_control/event
+	/// When do we start our event
+	var/start_time = 0
+	/// If we were created by a storyteller, here's a cost to refund in case.
+	var/cost
+	/// Whether we alerted admins about this schedule when it's close to being invoked.
+	var/alerted_admins = FALSE
+	/// Whether we are faking an occurence or not
+	var/fakes_occurence = TRUE
+	/// Whether this ignores event can run checks. If bussed by an admin, you want to ignore checks
+	var/ignores_checks
+	/// Whether the scheduled event will override the announcement change. If null it won't. TRUE = force yes. FALSE = force no.
+	var/announce_change
+
+/datum/scheduled_event/New(datum/round_event_control/passed_event, passed_time, passed_cost, passed_ignore, passed_announce)
+	. = ..()
+	event = passed_event
+	start_time = passed_time
+	cost = passed_cost
+	ignores_checks = passed_ignore
+	announce_change = passed_announce
+	/// Add a fake occurence to make the weightings/checks properly respect the scheduled event.
+	event.add_occurence()
+	fakes_occurence = TRUE
+
+/datum/scheduled_event/proc/remove_occurence()
+	if(fakes_occurence)
+		/// Remove the fake occurence if we still have it
+		event.subtract_occurence()
+		fakes_occurence = FALSE
+
+/// For admins who want to reschedule the event.
+/datum/scheduled_event/proc/reschedule(new_time)
+	start_time = new_time
+	alerted_admins = FALSE
+
+/datum/scheduled_event/proc/get_href_actions()
+	var/round_started = SSticker.HasRoundStarted()
+	if(round_started)
+		return "<a href='?src=[REF(src)];action=fire'>Fire</a> <a href='?src=[REF(src)];action=reschedule'>Reschedule</a> <a href='?src=[REF(src)];action=cancel'>Cancel</a> <a href='?src=[REF(src)];action=refund'>Refund</a></td>"
+	else
+		return "<a href='?src=[REF(src)];action=cancel'>Cancel</a>"
+
+/// Try and fire off the scheduled event
+/datum/scheduled_event/proc/try_fire()
+	/// Remove our fake occurence pre-emptively for the checks.
+	remove_occurence()
+
+	///If we can't spawn the scheduled event, refund it.
+	if(!ignores_checks && !event.can_spawn_event(FALSE)) //FALSE argument to ignore popchecks, to prevent scheduled events from failing from people dying/cryoing etc.
```

Comment:
can_spawn_events doesn't use popchecks as its first arg it uses player number replace with a high number for the same effect